internal/charmstore/migrations.go

```diff
@@ -222,3 +227,96 @@ func populateWriteACL(db StoreDatabase) error {
 	logger.Infof("%d base entities updated", counter)
 	return nil
 }
+
+func populatePromulgatedEntities(db StoreDatabase) error {
+	baseEntities := db.BaseEntities()
+	entities := db.Entities()
+
+	// 1. Update All Base Entities to have a promulgated value of the correct form.
+	_, err := baseEntities.UpdateAll(
+		bson.D{{"promulgated", true}},
+		bson.D{{"$set", bson.D{{"promulgated", 1}}}},
+	)
```

Comment:
I think the migration should be idempotent as much as possible, although they'll be changed before the existings entities are deleted so that will probably work.